### ISSUE TITLE
Existing token does not get deleted by terraform when removed or changed

### ISSUE THREAD

**Issue by ivnovi** [datastax/terraform-provider-astra#33]
### Terraform Version
v1.0.7

### Affected Resource
* astra_token

### Terraform Configuration Files
```hcl
terraform {
  required_providers {
    astra = {
      source  = "datastax/astra"
      version = "1.0.9"
    }
  }
}

resource "astra_database" "this" {
  name           = "database1"
  keyspace       = "keyspace1"
  cloud_provider = "aws"
  region         = "eu-central-1"
}

resource "astra_keyspace" "this" {
  name        = "keyspace1"
  database_id = resource.astra_database.this.id
}

resource "astra_role" "example" {
  role_name   = "keyspace1_example_role"
  description = "example"
  effect      = "allow"
  resources = [
    "drn:astra:org:${resource.astra_database.this.organization_id}:db:${resource.astra_database.this.id}:keyspace:${resource.astra_keyspace.this.name}:table:*"
  ]
  policy = [
    "db-table-create",
    "db-table-modify"
  ]
}

resource "astra_token" "example" {
  roles = [
    "${resource.astra_role.example.role_id}"
  ]
}
```

### Expected Behavior
When removed or changed existing token should be removed

### Actual Behavior
Existing token remains visible in Astra web console when removed from code. 
When token changed in code new one gets created without deleting the old one, so after change you have two instead of one.

### Steps to Reproduce
1. `terraform apply`
1. remove token in code
1. `terraform apply`

┆Issue is synchronized with this [Jira Task](https://datastax-oss.atlassian.net/browse/TERRA-1) by [Unito](https://www.unito.io)

**Comment by jdonenine** [comment]
### Notes
* We need to do a bit of investigation into the desired behavior from the terraform perspective.  Is applying a plan that implies a destruction of an internal resource forcing a destroy? Should it?
* It looks like the destroy/delete function exists within the provider, but it may not be trigger for some reason, or is failing when it is triggered

**Comment by emerkle826** [comment]
@ivnovi This should be fixed in version 1.0.11, which should be available [here](https://github.com/datastax/terraform-provider-astra/releases/tag/v1.0.11) shortly.

**Comment by ivnovi** [comment]
@emerkle826 looks good, thank you!
